src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -532,187 +897,241 @@ contract StrategyManagerUnitTests is Test, Utils {
+            sharesBefore[i] = strategyManager.stakerStrategyShares(staker, strategies[i]);
+            require(sharesBefore[i] == depositAmounts[i], "Staker has not deposited amount into strategy");
+            require(_isDepositedStrategy(staker, strategies[i]), "strategy should be deposited");
+        }
+        uint256 stakerStrategyListLengthBefore = strategyManager.stakerStrategyListLength(staker);
 
-        cheats.startPrank(staker);
-        cheats.expectRevert();
-        strategyManager.depositIntoStrategy(strategy, token, amount);
-        cheats.stopPrank();
+        uint256 numPoppedStrategies = 0;
+        uint256[] memory sharesAfter = new uint256[](3);
+        for (uint256 i = 0; i < 3; ++i) {
+            delegationManagerMock.removeShares(strategyManager, staker, strategies[i], sharesAmounts[i]);
+            sharesAfter[i] = strategyManager.stakerStrategyShares(staker, strategies[i]);
+            if (sharesAmounts[i] == depositAmounts[i]) {
+                ++numPoppedStrategies;
+                require(
+                    !_isDepositedStrategy(staker, strategies[i]),
+                    "strategy should not be part of staker strategy list"
+                );
+                require(sharesAfter[i] == 0, "sharesAfter != 0");
+            } else {
+                require(_isDepositedStrategy(staker, strategies[i]), "strategy should be part of staker strategy list");
+                require(
+                    sharesAfter[i] == sharesBefore[i] - sharesAmounts[i],
+                    "sharesAfter != sharesBefore - sharesAmounts"
+                );
+            }
+        }
+        require(
+            stakerStrategyListLengthBefore - numPoppedStrategies == strategyManager.stakerStrategyListLength(staker),
+            "stakerStrategyListLengthBefore - numPoppedStrategies != strategyManager.stakerStrategyListLength(staker)"
+        );
     }
+}
 
-    function test_depositIntoStrategyRevertsWhenStrategyDepositFunctionReverts() external {
-        // replace 'dummyStrat' with one that always reverts
-        dummyStrat = StrategyBase(address(new Reverter()));
-
-        // whitelist the strategy for deposit
-        cheats.startPrank(strategyManager.owner());
-        IStrategy[] memory _strategy = new IStrategy[](1);
-        _strategy[0] = dummyStrat;
-        cheats.expectEmit(true, true, true, true, address(strategyManager));
-        emit StrategyAddedToDepositWhitelist(dummyStrat);
-        strategyManager.addStrategiesToDepositWhitelist(_strategy);
-        cheats.stopPrank();
+contract StrategyManagerUnitTests_addShares is StrategyManagerUnitTests {
+    function test_Revert_DelegationManagerModifier() external {
+        DelegationManagerMock invalidDelegationManager = new DelegationManagerMock();
+        cheats.expectRevert(bytes("StrategyManager.onlyDelegationManager: not the DelegationManager"));
+        invalidDelegationManager.addShares(strategyManager, address(this), dummyStrat, 1);
+    }
```

Comment:
this test seems OK but it's structure is a bit unusual, in particular since it relies on the details of the 'DelegationManagerMock' contract. seems OK, mostly just pointing this out as it stood out to be as "different"